WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/builders/ScanStateListItemsBuilderTest.kt

```diff
@@ -146,43 +157,324 @@ class ScanStateListItemsBuilderTest : BaseUnitTest() {
+        )
+    }
+
+    @Test
+    fun `given idle state with not fixable threats, when items are built, items contain fix all threats button`() =
+        test {
+            val threats = listOf(threat.copy(baseThreatModel = baseThreatModel.copy(fixable = null)))
+            val scanStateModelWithNoFixableThreats = scanStateModelWithThreats.copy(threats = threats)
+
+            val scanStateItems = buildScanStateItems(scanStateModelWithNoFixableThreats)
+
+            assertThat(
+                scanStateItems.filterIsInstance(ActionButtonState::class.java)
+                    .map { it.text }
+            ).doesNotContain(
+                UiStringRes(R.string.threats_fix_all)
+            )
+        }
+
+    @Test
+    fun `given idle state with threats, when items are built, items contain clickable help text info in description`() =
+        test {
+            val clickableText = "clickable help text"
+            val descriptionWithClickableText = "description with $clickableText"
+            whenever(htmlMessageUtils.getHtmlMessageFromStringFormatResId(anyInt(), any()))
+                .thenReturn(descriptionWithClickableText)
+            whenever(resourceProvider.getString(R.string.scan_here_to_help)).thenReturn(clickableText)
+
+            val scanStateItems = buildScanStateItems(scanStateModelWithThreats)
+
+            val descriptionState = scanStateItems.filterIsInstance(DescriptionState::class.java).first()
+            assertThat(descriptionState.clickableTextsInfo?.first()).isEqualTo(
+                ClickableTextInfo(
+                    17,
+                    36,
+                    onHelpClickedMock
+                )
+            )
+        }
+
+    /* IDLE - THREATS NOT FOUND STATE */
+
+    @Test
+    fun `given idle no threats state, when items are built, items contain shield tick icon with green color`() =
+        test {
+            val scanStateItems = buildScanStateItems(scanStateModelWithNoThreats)
+
+            assertThat(scanStateItems.filterIsInstance(IconState::class.java).first()).isEqualTo(
+                IconState(
+                    icon = R.drawable.ic_shield_tick_white,
+                    colorResId = R.color.jetpack_green_40,
+                    sizeResId = R.dimen.scan_icon_size,
+                    marginResId = R.dimen.scan_icon_margin,
+                    contentDescription = UiStringRes(R.string.scan_state_icon)
+                )
+            )
+        }
+
+    @Test
+    fun `given idle no threats state, when items are built, items contain header for no threats found`() = test {
+        val scanStateItems = buildScanStateItems(model = scanStateModelWithNoThreats)
+
+        assertThat(scanStateItems.filterIsInstance(HeaderState::class.java).first()).isEqualTo(
+            HeaderState(UiStringRes(R.string.scan_idle_no_threats_found_title))
+        )
+    }
+
+    @Test
+    fun `given idle no threats state, when items are built, items contain scan again button `() = test {
```

Comment:
Minor (🔍): Consider fixing the typo in `again`. What you probably want is `now`, isn't it?

Reply:
Fixed in b309ef4